Automattic/wp-calypso
# A4A: Port the Jetpack Manage StoredCreditCard component to A4A.

This pull request ports the [StoredCreditCard](https://github.com/Automattic/wp-calypso/tree/trunk/client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2) component from Jetpack Manage to A4A. Most of the code is copy and paste, but there are also a few improvements to simplify some of the code.

<img width="859" alt="Screenshot 2024-03-06 at 7 24 39 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/8a0a956a-8955-4d2e-8b01-9d67a12098dd">


Closes https://github.com/Automattic/jetpack-genesis/issues/278

## Proposed Changes

* Port the [StoredCreditCard](https://github.com/Automattic/wp-calypso/tree/trunk/client/jetpack-cloud/sections/partner-portal/stored-credit-card-v2) component and its related components and hooks (e.g., dialog, confirmation, etc.).
* Mock mutation calls for both SetAsPrimaryCard and Delete operations.


## Testing Instructions

* Switch branch: `git checkout add/a4a-credit-card-component`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/purchases/payment-methods
* Confirm That the Credit cards listed work the same from https://cloud.jetpack.com/partner-portal/payment-methods
* Note that **SetAsPrimaryCard** and **Delete** operations are mocked.

## Pre-merge Checklist
